server/src/main/java/org/opensearch/gateway/remote/routingtable/RemoteIndexRoutingTable.java

```diff
@@ -9,92 +9,107 @@
+        super(clusterUUID, compressor);
+        this.index = null;
+        this.term = -1;
+        this.version = -1;
+        this.blobName = blobName;
+    }
 
-                int numberOfShardRouting = in.readVInt();
-                IndexRoutingTable.Builder indicesRoutingTable = IndexRoutingTable.builder(index);
-                for (int idx = 0; idx < numberOfShardRouting; idx++) {
-                    IndexShardRoutingTable indexShardRoutingTable = IndexShardRoutingTable.Builder.readFrom(in);
-                    indicesRoutingTable.addIndexShard(indexShardRoutingTable);
-                }
-                verifyCheckSum(in);
-                indexRoutingTable = indicesRoutingTable.build();
-            }
-        } catch (EOFException e) {
-            throw new IOException("Indices Routing table is corrupted", e);
-        }
+    public Index getIndex() {
+        return index;
     }
 
-    public IndexRoutingTable getIndexRoutingTable() {
-        return indexRoutingTable;
+    @Override
+    public BlobPathParameters getBlobPathParameters() {
+        return new BlobPathParameters(List.of(indexRoutingTable.getIndex().getUUID()), INDEX_ROUTING_FILE);
     }
 
-    /**
-     * Writes {@link IndexRoutingTable} to the given stream
-     * @param streamOutput output stream to write
-     * @throws IOException exception thrown on failing to write to stream.
-     */
     @Override
-    public void writeTo(StreamOutput streamOutput) throws IOException {
-        try {
-            BufferedChecksumStreamOutput out = new BufferedChecksumStreamOutput(streamOutput);
-            IndexRoutingTableHeader indexRoutingTableHeader = new IndexRoutingTableHeader(indexRoutingTable.getIndex().getName());
-            indexRoutingTableHeader.writeTo(out);
-            out.writeVInt(indexRoutingTable.shards().size());
-            for (IndexShardRoutingTable next : indexRoutingTable) {
-                IndexShardRoutingTable.Builder.writeTo(next, out);
-            }
-            out.writeLong(out.getChecksum());
-            out.flush();
-        } catch (IOException e) {
-            throw new IOException("Failed to write IndexRoutingTable to stream", e);
-        }
+    public String getType() {
+        return INDEX_ROUTING_TABLE;
     }
 
-    private void verifyCheckSum(BufferedChecksumStreamInput in) throws IOException {
-        long expectedChecksum = in.getChecksum();
-        long readChecksum = in.readLong();
-        if (readChecksum != expectedChecksum) {
-            throw new IOException(
-                "checksum verification failed - expected: 0x"
-                    + Long.toHexString(expectedChecksum)
-                    + ", got: 0x"
-                    + Long.toHexString(readChecksum)
+    @Override
+    public String generateBlobFileName() {
+        if (blobFileName == null) {
+            blobFileName = String.join(
+                DELIMITER,
+                getBlobPathParameters().getFilePrefix(),
```

Comment:
Do we use BlobPathParameter anywhere else? Otherwise this looks redundant since we are creating this object and then getting filePrefix which could have been directly supplied to this function.